client/web/compose/src/components/PageBlocks/RecordConfigurator.vue

```diff
@@ -42,11 +117,55 @@ export default {
 
   components: {
     FieldPicker,
+    VueSelect,
   },
 
   extends: base,
+
+  computed: {
+    documentationURL () {
+      // eslint-disable-next-line no-undef
+      const [year, month] = VERSION.split('.')
+      return `https://docs.cortezaproject.org/corteza-docs/${year}.${month}/integrator-guide/compose-configuration/index.html`
+    },
+
+    fieldOptions () {
+      const nonRequiredFields = this.module.fields.filter(f => !f.isRequired)
```

Comment:
We dont need to filter two times to do this. Also you should not allow all module fields, just use the ones that are selected in the above field picker!

The key is disabling the already selected ones and the required fields, there is a prop for that for vue-select (selectable prop)